io/disk/vfc-tests/vfc_unmap_map.py

```diff
@@ -0,0 +1,337 @@
+                self.vfc_dic["paths"] = self.get_paths(self.vfc_dic["vfc_client"])
+                self.log.info("complete list at paths: %s" % self.vfc_dic)
+                self.dic_list.append(self.vfc_dic)
+
+        self.log.info("complete list : %s" % self.dic_list)
+
+    def login(self, ipaddr, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        pxh = pxssh.pxssh()
+        # Work-around for old pxssh not having options= parameter
+        pxh.SSH_OPTS = pxh.SSH_OPTS + " -o 'StrictHostKeyChecking=no'"
+        pxh.SSH_OPTS = pxh.SSH_OPTS + " -o 'UserKnownHostsFile /dev/null' "
+        pxh.force_password = True
+
+        pxh.login(ipaddr, username, password)
+        pxh.sendline()
+        pxh.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        pxh.set_unique_prompt()
+        pxh.prompt(timeout=60)
+        self.pxssh = pxh
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on hmc %s", command)
+        con = self.pxssh
+        con.sendline(command)
+        con.expect("\n")  # from us
+        con.expect(con.PROMPT, timeout=timeout)
+        output = con.before.decode('utf-8').splitlines()
+        con.sendline("echo $?")
+        con.prompt(timeout)
+        return output
+
+    @staticmethod
+    def get_mcp_component(component):
+        '''
+        probes IBM.MCP class for mentioned component and returns it.
+        '''
+        for line in process.system_output('lsrsrc IBM.MCP %s' % component,
+                                          ignore_status=True, shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                return line.split()[-1].strip('{}\"')
+        return ''
+
+    @staticmethod
+    def get_partition_name(component):
+        '''
+        get partition name from lparstat -i
+        '''
+
+        for line in process.system_output('lparstat -i', ignore_status=True,
+                                          shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                return line.split(':')[-1].strip()
+        return ''
+
+    def test(self):
+        '''
+        Remove and add vfc interfaces Dynamically from HMC
+        '''
+        for _ in range(self.count):
+            for vfc_dic in self.dic_list:
+                self.vfchost_map_unmap("unmap", vfc_dic)
+                time.sleep(self.opp_sleep_time)
```

Comment:
why do we need sleep here ?
Also, why sleep only after unmap and not after map ?

Reply:
need to add the sleep even after map too. we need the sleep for sure but value can be reduced. will do

Reply:
I would say, `map()` should make the system stable so that you don't need sleep here.
And in `map()`, you should find a way to make it stable without sleep. sleep is not a good way to handle things, it is only a workaround.

Reply:
the status change will different amount of time in different distros
I had a thought of using wait functions but the status may change or may not chage due to some issue. hence again we have to provide the wait time there too, which as good as sleep time only. and I have one more functionality check here (fast_io_fail_tmo anddev_loss_tmo attribute check) for which I have to give sleep time for sure. so having sleep time in common will make the code simple. hence used.

So as per my knowledge, we have two way
1. use wait.wait_for(mpath_svc.status, timeout=10) and use sleep time for above attribute functionality check
2. use the sleep_time for both

let me know which you feel is better

for option 1 I have to change the verification functions to TRUE or FALSE to and need changes at multiple places. and if needed , have to write a separate function to handle it.

Reply:
@Naresh-ibm option sounds better